net/net/src/TSocket.cxx

```diff
@@ -103,7 +103,6 @@ TSocket::TSocket(TInetAddress addr, const char *service, Int_t tcpwindowsize)
                                         tcpwindowsize);
 
       if (fSocket != -1) {
-         R__LOCKGUARD(gROOTMutex);
```

Comment:
Are all locks around `gROOT->GetListOf*()` now redundant? I suspect that we still have many other places to fix then, to avoid deadlocks, no?

Reply:
> Are all locks around gROOT->GetListOf*() now redundant? 

Yes but the 'right' solution is not necessarily to remove them.  For example code like:
```
R__LOCKGUARD(gROOTMutex);
gROOT->GetListOfXYZ()->Add();
OtherGlobalAction();
```
is better off acquiring the lock outside the collection rather than taking it twice.  So no easy solution.

>  to avoid deadlocks, no

There is no risk of dead-lock as this the same lock that it taken either way. gROOTMutex, gInterpreterMutex and ROOT::gCoreMutex points to the same object.  gROOTMutex, gInterpreterMutex are deprecated and should be gradually removed.  They are kept in places until the code has been reviewed to see if it should be replaced by taking the read lock or the write lock or something else.  i.e. they are now a marker or non reviewed code.